### PR TITLE
PROTON-2137: Removing ssl init from ssl_server_options default constructor

### PR DESCRIPTION
**Pull request by rabih-mourad**

### CODE REVIEW

**Comment by rabih-mourad** [comment]
Weird, I just renamed the commit and travis-ci failed...

File: cpp/src/connection_options.cpp [hunk 1]
```diff
@@ -175,7 +175,8 @@ class connection_options::impl {
             }
         } else if (!client && ssl_server_options.set) {
             pn_ssl_t *ssl = pn_ssl(pnt);
-            if (pn_ssl_init(ssl, ssl_server_options.value.impl_->pn_domain(), NULL)) {
+            pn_ssl_domain_t* ssl_domain = ssl_server_options.value.impl_ ? ssl_server_options.value.impl_->pn_domain() : NULL;
+            if (pn_ssl_init(ssl, ssl_domain, NULL)) {
```
Comment (astitcher) [inline cpp/src/connection_options.cpp]: This change is wrong unfortunately.
the comparison with the client case is misleading. `pn_ssl_init(ssl, NULL, NULL)` will create an ssl connection with the default domain - however the default domain is a *client* domain so you can't do this if the connection is a server.
So you need to replace he NULL with something like `pn_ssl_domain(PN_SSL_MODE_SERVER)`.

Comment (astitcher) [reply cpp/src/connection_options.cpp]: It is a shame that this wasn't picked up by any tests - I thought we had some ssl server tests.

Comment (rabih-mourad) [reply cpp/src/connection_options.cpp]: @astitcher  I replaced NULL by pn_ssl_domain(PN_SSL_MODE_SERVER) like you mentioned

**Comment by astitcher** [comment]
@rabih-mourad Could you squash this into a single commit please - then I can rebase and merge it.

**Comment by rabih-mourad** [comment]
@astitcher squash done. Will we add a test in this pull request or we will do another one for it?

**Comment by astitcher** [comment]
I'm happy to commit this as is. I think testing c++ ssl servers is probably a different issue.

**Comment by jdanekrh** [comment]
That travis failure was unrelated to the PR. Something changed either on Travis or in Brew (brew.sh) so that openssl .dynlib is no longer found when `PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig` is set.

I think I know what to do about it.

**Pull request merged**